js/src/builders.ts

```diff
@@ -10,12 +10,23 @@ type LastMessageOpts<T extends keyof MessageTypes> = {
   last: TbDEXMessage<T>
 }
 
+type ReplyMessageOpts<T extends keyof MessageTypes> = {
+  reply: TbDEXMessage<T>
+}
+
 export type CreateMessageOpts<T extends keyof MessageTypes, U extends keyof MessageTypes> = XOR<NewThreadOpts, LastMessageOpts<U>> & {
   type: T,
   body: MessageType<T>,
 }
 
-export function createMessage<T extends keyof MessageTypes, U extends keyof MessageTypes>(opts: CreateMessageOpts<T, U>): TbDEXMessage<T> {
+export type CreateMessageOpts2<T extends keyof MessageTypes, U extends keyof MessageTypes, X extends keyof MessageTypes> =
+XOR<NewThreadOpts, ReplyMessageOpts<U>> & {
+  type: T,
+  body: MessageType<T>,
+  last?: TbDEXMessage<X>
```

Comment:
In which situations do you pass in `last` vs `reply`? And in which situations would you pass in _both_?

Reply:
ok we have a smol matrix here

no last, no reply: initial message (i.e. RFQ) - i expect you to pass in to and from

yes last, no reply: replying to a thread where last message is from you (i.e. OrderStatus, Close?) - i expect you to pass in to and from

no last, yes reply:... i can't think of an example here.

yes last, yes reply: replying to a thread where last message is NOT from you (i.e. PFI sends Quote in response to having received RFQ) - i'll fill in the to and from for you